src/ImageSharp/Drawing/Processors/DrawPathProcessor.cs

```diff
@@ -0,0 +1,160 @@
+            : this(pen, shape.ToArray())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DrawPathProcessor{TColor, TPacked}"/> class.
+        /// </summary>
+        /// <param name="pen">The pen.</param>
+        /// <param name="paths">The paths.</param>
+        public DrawPathProcessor(IPen<TColor, TPacked> pen, params IPath[] paths)
+        {
+            this.paths = paths;
+            this.pen = pen;
+
+            if (paths.Length != 1)
+            {
+                var maxX = paths.Max(x => x.Bounds.Right);
+                var minX = paths.Min(x => x.Bounds.Left);
+                var maxY = paths.Max(x => x.Bounds.Bottom);
+                var minY = paths.Min(x => x.Bounds.Top);
+
+                this.region = new RectangleF(minX, minY, maxX - minX, maxY - minY);
+            }
+            else
+            {
+                this.region = paths[0].Bounds;
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override void OnApply(ImageBase<TColor, TPacked> source, Rectangle sourceRectangle)
+        {
+            using (IPenApplicator<TColor, TPacked> applicator = this.pen.CreateApplicator(this.region))
+            {
+                var rect = RectangleF.Ceiling(applicator.RequiredRegion);
+
+                int polyStartY = rect.Y - PaddingFactor;
+                int polyEndY = rect.Bottom + PaddingFactor;
+                int startX = rect.X - PaddingFactor;
+                int endX = rect.Right + PaddingFactor;
+
+                int minX = Math.Max(sourceRectangle.Left, startX);
+                int maxX = Math.Min(sourceRectangle.Right, endX);
+                int minY = Math.Max(sourceRectangle.Top, polyStartY);
+                int maxY = Math.Min(sourceRectangle.Bottom, polyEndY);
+
+                // Align start/end positions.
+                minX = Math.Max(0, minX);
+                maxX = Math.Min(source.Width, maxX);
+                minY = Math.Max(0, minY);
+                maxY = Math.Min(source.Height, maxY);
+
+                // Reset offset if necessary.
+                if (minX > 0)
+                {
+                    startX = 0;
+                }
+
+                if (minY > 0)
+                {
+                    polyStartY = 0;
+                }
+
+                using (PixelAccessor<TColor, TPacked> sourcePixels = source.Lock())
+                {
+                    Parallel.For(
+                    minY,
+                    maxY,
+                    this.ParallelOptions,
+                    y =>
+                    {
+                        int offsetY = y - polyStartY;
+
+                        for (int x = minX; x < maxX; x++)
+                        {
+                            int offsetX = x - startX;
+
+                            var dist = this.paths.Select(p => p.Distance(offsetX, offsetY)).OrderBy(p => p.DistanceFromPath).First();
```

Comment:
This line is an absolute performance killer. For each pixel we are creating two arrays. One for Select, one for OrderBy. We need to find a way to avoid this or the memory usage will be incredible.